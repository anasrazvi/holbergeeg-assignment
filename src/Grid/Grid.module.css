.gridContainer{
    overflow: auto;
    height: calc(100% - (42px));
  }