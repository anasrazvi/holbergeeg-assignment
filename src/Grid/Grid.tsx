import {observer} from 'mobx-react';
import {Header} from '../Header/Header';
import {Rows} from '../Rows/Rows';
import {storeProps} from '../Types/Interfaces'
import GridCSS from './Grid.module.css'

export const Grid: React.FC<storeProps> = observer(({store}) => {
    return (
        <div className={GridCSS.gridContainer}>
          <Header store={store}></Header>
          <Rows store={store}></Rows>
        </div>
    )
  });