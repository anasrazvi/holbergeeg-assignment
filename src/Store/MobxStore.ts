import { autorun, makeAutoObservable } from "mobx";
import {columnsMeta, ColumnsMetaInformation, StudyData, DataEntity, StudyDataMapped } from '../Types/Interfaces';

export class ObservableStoreImpl {
    //Initialize the store
    metaData: ColumnsMetaInformation = {
        columnsMetaInformation: []
    }
    studyData: StudyData = {
      data: []
    }
    studyDataMapped: StudyDataMapped = {
      data: new Map<number, DataEntity>()
    }
    isLoaded: boolean = true;
    constructor() {
      makeAutoObservable(this)
      autorun(() => console.log(this.metaData));
    }
  
    //Setters
    setMetaDataSuccess(data: columnsMeta[]) {
      data.sort((a,b) => {return (a.order || 0) - (b.order || 0)})
      this.metaData.columnsMetaInformation = data;
    }

    setStudyDataSuccess(data: DataEntity[]){
      /*const mappedData: Map<number, DataEntity> = new Map(data.map(study => [study.studyId?.primaryValue || study.studyId?.secondaryValue, study] as [number, DataEntity]))
      mappedData.forEach((value: DataEntity, key: number) => {
        this.studyDataMapped.data.set(key, value);
      })*/
      this.studyData.data = data;
    }
    
    setVisibility(colIndex: number, isVisible: boolean){
      if(this.metaData.columnsMetaInformation){
        this.metaData.columnsMetaInformation[colIndex].isVisible = isVisible; 
      }
    }

    setLoader(isLoading: boolean){
      this.isLoaded = isLoading;
    }

    //Getters
    get Loader(){
      return this.isLoaded;
    }

    get Metadata() {
       return { metaData: this.metaData.columnsMetaInformation }
    }

    get StudyData(){
      return { 
        studyData: this.studyData.data
      }
    }
  }
  
  export const observableStore = new ObservableStoreImpl();