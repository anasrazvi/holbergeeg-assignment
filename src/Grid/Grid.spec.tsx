import renderer from 'react-test-renderer';
import {Grid} from '../Grid/Grid';
import { ObservableStoreImpl } from '../Store/MobxStore';
import {data} from '../StudyData';

const store = new ObservableStoreImpl();
store?.setMetaDataSuccess(data.GetMetadata().columnsMetaInformation);
store?.setStudyDataSuccess(data.GetStudyList().data);

describe("Grid Component", () => {
    test('renders grid component', () => {
        const grid = renderer.create(<Grid store={store}></Grid>).toJSON();
        expect(grid).toMatchSnapshot();
    });
});