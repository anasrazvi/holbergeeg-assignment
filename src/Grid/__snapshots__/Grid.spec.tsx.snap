// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Component renders grid component 1`] = `
<div
  className="gridContainer"
>
  <div
    className="header"
  >
    <div
      className="col"
    >
      <strong>
        Study start
      </strong>
    </div>
    <div
      className="col"
    >
      <strong>
        Study length
      </strong>
    </div>
    <div
      className="col"
    >
      <strong>
        Study type
      </strong>
    </div>
    <div
      className="col"
    >
      <strong>
        Study status
      </strong>
    </div>
    <div
      className="col"
    >
      <strong>
        Name
      </strong>
    </div>
    <div
      className="col"
    >
      <strong>
        Patient ID
      </strong>
    </div>
    <div
      className="col"
    >
      <strong>
        Date of birth
      </strong>
    </div>
    <div
      className="col"
    >
      <strong>
        Age at the time of study
      </strong>
    </div>
  </div>
  <div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        2021-01-15 13:29
      </div>
      <div
        className="col"
      >
        30 minutes 
      </div>
      <div
        className="col"
      >
        Sleep deprived EEG
      </div>
      <div
        className="col"
      >
        <div
          className="studyStatus"
        >
          New
        </div>
      </div>
      <div
        className="col"
      >
        TESTr, Test Test
      </div>
      <div
        className="col"
      >
        TEST1
      </div>
      <div
        className="col"
      >
        1990-12-28
      </div>
      <div
        className="col"
      >
        30 years
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        2021-01-15 13:29
      </div>
      <div
        className="col"
      >
        30 minutes 
      </div>
      <div
        className="col"
      >
        Standard EEG
      </div>
      <div
        className="col"
      >
        <div
          className="studyStatus"
        >
          New
        </div>
      </div>
      <div
        className="col"
      >
        Taylor, Billy Lee
      </div>
      <div
        className="col"
      >
        IdentityString41
      </div>
      <div
        className="col"
      >
        1990-12-28
      </div>
      <div
        className="col"
      >
        30 years
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        2021-01-15 13:29
      </div>
      <div
        className="col"
      >
        30 minutes 
      </div>
      <div
        className="col"
      >
        Long term monitoring (LTM)
      </div>
      <div
        className="col"
      >
        <div
          className="studyStatus"
        >
          New
        </div>
      </div>
      <div
        className="col"
      >
        Taylor, Billy Lee
      </div>
      <div
        className="col"
      >
        IdentityString24
      </div>
      <div
        className="col"
      >
        1990-12-28
      </div>
      <div
        className="col"
      >
        30 years
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        2020-09-16 13:29
      </div>
      <div
        className="col"
      >
        30 minutes 
      </div>
      <div
        className="col"
      >
        Intraoperative monitoring
      </div>
      <div
        className="col"
      >
        <div
          className="studyStatus"
        >
          New
        </div>
      </div>
      <div
        className="col"
      >
        Jones, James Beth
      </div>
      <div
        className="col"
      >
        IdentityString23
      </div>
      <div
        className="col"
      >
        2020-06-14
      </div>
      <div
        className="col"
      >
        3 months, 4 weeks
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        2020-08-25 14:30
      </div>
      <div
        className="col"
      >
        1 hour 20 minutes 
      </div>
      <div
        className="col"
      >
        Ambulatory recording
      </div>
      <div
        className="col"
      >
        <div
          className="studyStatus"
        >
          New
        </div>
      </div>
      <div
        className="col"
      >
        RODRIGUEZAl, fred Judith
      </div>
      <div
        className="col"
      >
        IdentityString10
      </div>
      <div
        className="col"
      >
        1996-09-15
      </div>
      <div
        className="col"
      >
        23 years
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        2020-08-25 14:30
      </div>
      <div
        className="col"
      >
        1 day 6 hours 
      </div>
      <div
        className="col"
      >
        Wada test
      </div>
      <div
        className="col"
      >
        <div
          className="studyStatus"
        >
          New
        </div>
      </div>
      <div
        className="col"
      >
        WILSON, Barbara Carolyn
      </div>
      <div
        className="col"
      >
        IdentityString15
      </div>
      <div
        className="col"
      >
        2020-01-20
      </div>
      <div
        className="col"
      >
        7 months
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        2020-08-25 14:30
      </div>
      <div
        className="col"
      >
        1 day 
      </div>
      <div
        className="col"
      >
        Ambulatory recording
      </div>
      <div
        className="col"
      >
        <div
          className="studyStatus"
        >
          New
        </div>
      </div>
      <div
        className="col"
      >
        JONES, Annie bilaspur
      </div>
      <div
        className="col"
      >
        IdentityString13
      </div>
      <div
        className="col"
      >
        1959-05-03
      </div>
      <div
        className="col"
      >
        61 years
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        2020-08-25 14:30
      </div>
      <div
        className="col"
      >
        1 day 
      </div>
      <div
        className="col"
      >
        ICU
      </div>
      <div
        className="col"
      >
        <div
          className="studyStatus"
        >
          New
        </div>
      </div>
      <div
        className="col"
      >
        RODRIGUEZ, Karen Janet
      </div>
      <div
        className="col"
      >
        IdentityString11
      </div>
      <div
        className="col"
      >
        1952-08-14
      </div>
      <div
        className="col"
      >
        68 years
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        2020-08-25 14:30
      </div>
      <div
        className="col"
      >
        1 day 
      </div>
      <div
        className="col"
      >
        Ambulatory recording
      </div>
      <div
        className="col"
      >
        <div
          className="studyStatus"
        >
          New
        </div>
      </div>
      <div
        className="col"
      >
        LEE, Larry Mary
      </div>
      <div
        className="col"
      >
        IdentityString21
      </div>
      <div
        className="col"
      >
        2004-06-16
      </div>
      <div
        className="col"
      >
        16 years
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        2020-08-25 14:30
      </div>
      <div
        className="col"
      >
        1 day 1 hour 5 minutes 
      </div>
      <div
        className="col"
      >
        Sleep deprived EEG
      </div>
      <div
        className="col"
      >
        <div
          className="studyStatus"
        >
          New
        </div>
      </div>
      <div
        className="col"
      >
        TAYLOR, Mary MARTIN
      </div>
      <div
        className="col"
      >
        IdentityString1
      </div>
      <div
        className="col"
      >
        1968-12-31
      </div>
      <div
        className="col"
      >
        51 years
      </div>
    </div>
  </div>
</div>
`;
